core/sdk-core/src/main/java/software/amazon/awssdk/core/sync/RequestBody.java

```diff
@@ -129,6 +129,8 @@ public static RequestBody fromFile(File file) {
      * 128 KiB. If you need more control, use {@link #fromContentProvider(ContentStreamProvider, long, String)} or
      * {@link #fromContentProvider(ContentStreamProvider, String)}.
      * <p>
+     * If a content length smaller than the actual size of the object is set, the client will truncate the stream to the
+     * specified content length and only send exactly the number of bytes equal to the content length.
      *
      * @param inputStream   Input stream to send to the service. The stream will not be closed by the SDK.
      * @param contentLength Content length of data in input stream.
```

Comment:
It will be of direct help if we mention this right next to @param contentLength , WDYT ?
```
 * @param inputStream   Input stream to send to the service. The stream will not be closed by the SDK.
 * @param contentLength Content length of data in input stream. 
 *                      If a content length smaller than the actual size of the 
 *                      object is set, the client will truncate the stream to the specified content length and only send 
 *                      exactly the number of bytes equal to the content length.
 * @return RequestBody instance.
 */


```